src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -253,18 +259,39 @@ private void removeCGFromMonitors(CertificateGenerator gen) {
         cisShadowMonitor.removeFromMonitor(gen);
     }
 
+    /**
+     * Configures the KeyStore to use a certificates provided from the CA configuration.
+     *
+     * @throws KeyStoreException if the stored key is not a private key
+     * @throws Exception if failing to get the keyPair or certificateChain
+     * @throws IllegalArgumentException if configuration is invalid for CA
+     */
+    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+    public void configureCustomCA() throws KeyStoreException, Exception {
```

Comment:
The reason for suppressing warnings is we use the `RetryUtils` which under the hood just throw plain `Exceptions` we would have to update it if we want to avoid doing this.

Reply:
This method should hide that complexity. The only exception types thrown by retryutil are the exceptions which can be thrown by the callable which it is retrying. You can simply extract out the true exception type and rethrow the real exception.